### PR TITLE
Fixing Oculus Mixed Reality Capture Camera pose

### PR DESCRIPTION
**Pull request by fabio914**
## Changes

This is an attempt to solve https://github.com/icosa-gallery/open-brush/issues/44 without modifying the Oculus Integration Package.

 * This PR fixes the issues with the Oculus Mixed Reality camera positions because of the different units used by Open Brush (decimeters instead of meters).

This PR is similar to https://github.com/icosa-gallery/open-brush/pull/70, however, it doesn't have the changes for Reality Mixer.

cc @mikeage @mikeskydev

### CODE REVIEW

File: Assets/Scripts/MRCCameraUpdate.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        // Converting position from meters to decimeters (unit used by Open Brush)
+        cameraStagePoseInUnits.position *= TiltBrush.App.METERS_TO_UNITS;
+
+        // Workaround to fix the OVRExtensions.ToWorldSpacePose() and OVRComposition.ComputeCameraWorldSpacePose()
+        // when computing the Mixed Reality foreground and background camera positions.
+        OVRPose headPose = OVRPose.identity;
+
+        Vector3 pos;
+        Quaternion rot;
+        if (OVRNodeStateProperties.GetNodeStatePropertyVector3(Node.Head, NodeStatePropertyType.Position, OVRPlugin.Node.Head, OVRPlugin.Step.Render, out pos))
```
Comment (TimAidley) [inline Assets/Scripts/MRCCameraUpdate.cs:52]: Nit: If we are to stay in line with Tilt Brush's code style, this line and line 54 should be split to stay within 100 chars long.

**Review by TimAidley — APPROVED** [review]
I'm excited to see this one in! I had a minor nit about line length.

File: Assets/Scripts/MRCCameraUpdate.cs (hunk 1 above)
Comment (fabio914) [reply Assets/Scripts/MRCCameraUpdate.cs]: Thank you @TimAidley. I've just updated the code style for this class to keep it consistent with the project, please let me know if this is ok.

Comment (TimAidley) [reply Assets/Scripts/MRCCameraUpdate.cs]: Looks good!

**Comment by mikeskydev** [comment]
Good idea to split them off, thanks! Will give this a test tonight and get it in.

**Comment by mikeskydev** [comment]
Gave this a shot with my webcam but was struggling to get a proper calibration going, will try tomorrow morning on our full MR setup. However, I did notice that the world camera capture seemed to move when I titled my head side to side, which probably shouldn't be happening with MR?

**Comment by fabio914** [comment]
> but was struggling to get a proper calibration going

The Oculus MRC Calibration app is freezing for me from time to time when I try to calibrate. Were you experiencing something similar? This started happening for me after my Quest started using version 27 of the system.

> However, I did notice that the world camera capture seemed to move when I tilted my head side to side, which probably shouldn't be happening with MR?

I tested again here, and I also noticed some movement, although it's very hard to notice (see the video below). My guess is that this happens because this update with the fix is out of sync with the MRC update (probably 1 or more frames apart), so the matrix for the head position that we're using here doesn't match exactly the one being used when the MRC camera position is being computed.

https://user-images.githubusercontent.com/2430631/113493920-c0e65000-94db-11eb-8cb9-d58f9a2a5336.mp4

**Review by mikeskydev — APPROVED** [review]
Got it working great, just needed a better calibration! Thanks for sorting this!

**Pull request merged**